define-helpers/define-helpers.js

```diff
@@ -7,79 +7,87 @@ var canEvent = require("can-event");
 
 
 var hasMethod = function(obj, method){
-    return obj && typeof obj === "object" && (method in obj);
+	return obj && typeof obj === "object" && (method in obj);
 };
 
 var defineHelpers = {
-    extendedSetup: function(props){
-        assign(this, props);
-    },
-    toObject: function(map, props, where, Type){
-        if(props instanceof Type) {
-            props.each(function(value, prop){
-                where[prop] = value;
-            });
-            return where;
-        } else {
-            return props;
-        }
-    },
-    defineExpando: function(map, prop, value) {
-        // first check if it's already a constructor define
-        var constructorDefines = map._define.definitions;
-        if(constructorDefines && constructorDefines[prop]) {
-            return;
-        }
-        // next if it's already on this instances
-        var instanceDefines = map._instanceDefinitions;
-        if(!instanceDefines) {
-            instanceDefines = map._instanceDefinitions = {};
-        }
-        if(!instanceDefines[prop]) {
-            var defaultDefinition = map._define.defaultDefinition || {type: define.types.observable};
-            define.property(map, prop, defaultDefinition, {},{});
-            // possibly convert value to List or DefineMap
-            map._data[prop] = defaultDefinition.type ? defaultDefinition.type(value) : define.types.observable(value);
-            instanceDefines[prop] = defaultDefinition;
-            canBatch.start();
-            canEvent.dispatch.call(map, {
-                type: "__keys",
-                target: map
-            });
-            if(map._data[prop] !== undefined) {
-                canEvent.dispatch.call(map, {
-                    type: prop,
-                    target: map
-                },[map._data[prop], undefined]);
-            }
-            canBatch.stop();
-            return true;
-        }
-    },
-    // ## getValue
+	extendedSetup: function(props){
+		assign(this, props);
+	},
+	toObject: function(map, props, where, Type){
+		if(props instanceof Type) {
+			props.each(function(value, prop){
+				where[prop] = value;
+			});
+			return where;
+		} else {
+			return props;
+		}
+	},
+	removeSpecialKeys: function(map) {
+		if(map) {
```

Comment:
see https://github.com/canjs/can-define/blob/improve-assignment/map/map.js#L32
https://github.com/canjs/can-define/blob/improve-assignment/map/map.js#L97
https://github.com/canjs/can-define/blob/improve-assignment/map/map.js#L125

The helper is called on the result of toObject and on a shallow clone of the map arg.  Because it had two different call cases, I opted to make the function not-smart and resolve that concern before calling it.